app/services/feeds/optimized.rb

```diff
@@ -0,0 +1,279 @@
+    def top_articles_by_timeframe(timeframe:)
+      published_articles_by_tag.where("published_at > ?", Timeframer.new(timeframe).datetime)
+        .order(score: :desc).page(@page).per(@number_of_articles)
+    end
+
+    def latest_feed
+      published_articles_by_tag.order(published_at: :desc)
+        .where("featured_number > ? AND score > ?", 1_449_999_999, -20)
+        .page(@page).per(@number_of_articles)
+    end
+
+    def default_home_feed_and_featured_story(user_signed_in: false, ranking: true)
+      featured_story, hot_stories = globally_hot_articles(user_signed_in)
+      hot_stories = rank_and_sort_articles(hot_stories) if @user && ranking
+      [featured_story, hot_stories]
+    end
+
+    # Test variation: Base
+    def default_home_feed(user_signed_in: false)
+      _featured_story, stories = default_home_feed_and_featured_story(user_signed_in: user_signed_in, ranking: true)
+      stories
+    end
+
+    # Test variation: More random
+    def default_home_feed_with_more_randomness_experiment
+      @randomness = 7
+      _featured_story, stories = default_home_feed_and_featured_story(user_signed_in: true)
+      stories
+    end
+
+    # Test variation: tags make bigger impact
+    def more_tag_weight_experiment
+      @tag_weight = 2
+      _featured_story, stories = default_home_feed_and_featured_story(user_signed_in: true)
+      stories
+    end
+
+    def more_tag_weight_more_random_experiment
+      @tag_weight = 2
+      @randomness = 7
+      _featured_story, stories = default_home_feed_and_featured_story(user_signed_in: true)
+      stories
+    end
+
+    # Test variation: Base half the time, more random other half. Varies on impressions.
+    def mix_default_and_more_random_experiment
+      if rand(2) == 1
+        default_home_feed(user_signed_in: true)
+      else
+        default_home_feed_with_more_randomness_experiment
+      end
+    end
+
+    # Test variation: the more comments a post has, the higher it's rated!
+    def more_comments_experiment
+      @comment_weight = 2
+      _featured_story, stories = default_home_feed_and_featured_story(user_signed_in: true)
+      stories
+    end
+
+    # Test variation: the more comments a post has, the higher it's rated!
+    def more_comments_medium_weight_experiment
+      @comment_weight = 0.5
+      _featured_story, stories = default_home_feed_and_featured_story(user_signed_in: true)
+      stories
+    end
+
+    # Test variation: the more comments a post has, the higher it's rated!
+    def more_comments_minimal_weight_experiment
+      @comment_weight = 0.2
+      _featured_story, stories = default_home_feed_and_featured_story(user_signed_in: true)
+      stories
+    end
+
+    def more_experience_level_weight_experiment
+      @experience_level_weight = 3
+      _featured_story, stories = default_home_feed_and_featured_story(user_signed_in: true)
+      stories
+    end
+
+    def mix_of_everything_experiment
```

Comment:
We have been running these experiments for a while now. I think its time to determine which one or few are the most ideal and stick with those. That will allow us to hon in and optimize for those feed cases. What is the plan for doing that?

Reply:
I also am curious if we have any metrics around these experiments; if not, do we even have enough information/data to know which of these actually _sticks_?

Reply:
@benhalpern what is the plan going forward for these experiments?